replication-api/src/main/java/com/connexta/replication/queue/Task.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a task that was queued for later processing by a worker.
+ *
+ * <p>When a task is first queued, its total attempts counter will report <code>0</code> and its
+ * queued time and original queued time will be set to the same time (i.e. the current time when the
+ * task is queued via one of {@link Queue#offer(TaskInfo)}), {@link Queue#offer(TaskInfo, long,
+ * TimeUnit)}, or {@link Queue#put(TaskInfo)}. If the task fails and a decision is made to put it
+ * back into the queue via one of {@link #offerForRetry(PriorityLevel)}, {@link
+ * #offerForRetry(PriorityLevel, long, TimeUnit)}, or {@link #putForRetry(PriorityLevel)}, a new
+ * task will be created with its total attempts counter increased by <code>1</code>. The original
+ * queued time will be copied from this task and its queued time will be set to the current time
+ * when the new created task will be queued.
+ */
+public interface Task extends TaskInfo {
+  /**
+   * Gets the queue from which this task was retrieved.
+   *
+   * @return the queue from which this task was retrieved
+   */
+  public Queue getQueue();
+
+  /**
+   * Gets the time when the first attempted task was queued.
+   *
+   * @return the time when the first attempted task was queued in milliseconds from the epoch
+   */
+  public long getOriginalQueuedTime();
+
+  /**
+   * Gets the time when the task was queued. This will be the same time as reported by {@link
+   * #getOriginalQueuedTime()} if the task has not yet been attempted ({@link #getTotalAttempts()}
+   * returns <code>0</code>).
+   *
+   * @return the time when this task was queued in milliseconds from the epoch
+   */
+  public long getQueuedTime();
+
+  /**
+   * Gets the total number of times this particular task has been attempted. When a task fails for a
+   * reason that doesn't preclude it from being re-tried and it gets re-queued by the worker, its
```

Comment:
This re-queuing is confusing me. What reason would a worker choose to evict a Task from the queue other than a success and why does a worker get to make that decision?